modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2445,50 +2448,112 @@ private CompletableFuture<BinaryRow> resolveWriteIntentAsync(
             HybridTimestamp timestamp,
             Supplier<BinaryRow> lastCommitted
     ) {
-        return resolveTxState(
-                new TablePartitionId(readResult.commitTableId(), readResult.commitPartitionId()),
-                readResult.transactionId(),
-                timestamp
-        ).thenApply(readLastCommitted -> {
-            if (readLastCommitted) {
-                return lastCommitted.get();
-            } else {
-                return readResult.binaryRow();
-            }
-        });
+        return resolveTxState(readResult, timestamp)
+                .thenApply(transactionReadable -> {
+                    // The value of `transactionReadable` is true - when transaction is committed and commit time <= read time.
+                    if (transactionReadable) {
+                        return readResult.binaryRow();
+                    } else {
+                        // Otherwise the transaction is either in progress, or aborted, so look for the last committed value.
+                        return lastCommitted.get();
+                    }
+                });
     }
 
     /**
-     * Resolve the actual tx state.
+     * Schedules an async cleanup action for the given write intent.
      *
-     * @param commitGrpId Commit partition id.
      * @param txId Transaction id.
+     * @param rowId Id of a row that we want to clean up.
+     * @param meta Resolved transaction state.
+     */
+    private void scheduleTransactionRowAsyncCleanup(UUID txId, RowId rowId, TransactionMeta meta) {
+        TxState txState = meta.txState();
+
+        HybridTimestamp commitTimestamp = meta.commitTimestamp();
+
+        // Add the resolved row to the set of write intents the transaction created.
+        // If the volatile state was lost on restart, we'll have a single item in that set,
+        // otherwise the set already contains this value.
+        storageUpdateHandler.handleWriteIntentRead(txId, rowId);
```

Comment:
Why do not want to  create a method with four parameters?
handleTransactionCleanup(
  RowId pendingRowId, 
  boolean commit,
  HybridTimestamp commitTimestamp, 
  Runnable onApplication
)